Mathlib/Algebra/PUnitInstances.lean

```diff
@@ -50,37 +52,57 @@ theorem mul_eq : x * y = star :=
   rfl
 #align punit.mul_eq PUnit.mul_eq
 
--- `sub_eq` simplifies `punit.sub_eq`, but the latter is eligible for `dsimp`
-@[simp, nolint simp_nf, to_additive]
+-- `sub_eq` simplifies `PUnit.sub_eq`, but the latter is eligible for `dsimp`
+@[simp, nolint simpNF, to_additive]
 theorem div_eq : x / y = star :=
   rfl
 #align punit.div_eq PUnit.div_eq
 
--- `neg_eq` simplifies `punit.neg_eq`, but the latter is eligible for `dsimp`
-@[simp, nolint simp_nf, to_additive]
+-- `neg_eq` simplifies `PUnit.neg_eq`, but the latter is eligible for `dsimp`
+@[simp, nolint simpNF, to_additive]
 theorem inv_eq : x⁻¹ = star :=
   rfl
 #align punit.inv_eq PUnit.inv_eq
 
-instance : CommRing PUnit := by
-  refine' { PUnit.commGroup, PUnit.addCommGroup with natCast := fun _ => PUnit.unit.. } <;>
-      intros <;>
-    exact Subsingleton.elim _ _
-
-instance : CancelCommMonoidWithZero PUnit := by
-  refine' { PUnit.commRing with .. } <;> intros <;> exact Subsingleton.elim _ _
-
-instance : NormalizedGcdMonoid PUnit := by
-  refine' {
-          gcd := fun _ _ => star
-          lcm := fun _ _ => star
-          normUnit := fun x => 1
-          gcd_dvd_left := fun _ _ => ⟨star, Subsingleton.elim _ _⟩
-          gcd_dvd_right := fun _ _ => ⟨star, Subsingleton.elim _ _⟩
-          dvd_gcd := fun _ _ _ _ _ => ⟨star, Subsingleton.elim _ _⟩
-          gcd_mul_lcm := fun _ _ => ⟨1, Subsingleton.elim _ _⟩.. } <;>
-      intros <;>
-    exact Subsingleton.elim _ _
+instance commRing: CommRing PUnit where
```

Comment:
You probably want this to inherit from the associated `CommGroup` and `AddCommGroup` instances (it's unlikely that there exists a change someone might make which would make them diverge, but better safe than sorry).  So something like
```lean
  { PUnit.CommGroup, PUnit.AddCommGroup with
    left_distrib := by intros; rfl
    right_distrib := by intros; rfl }
```

Reply:
done -- thanks for syntax!